ext/ffi/lib.rs

```diff
@@ -326,6 +357,14 @@ impl NativeValue {
       NativeType::Pointer | NativeType::Buffer | NativeType::Function => {
         Arg::new(&self.pointer)
       }
+      NativeType::Struct(_) => {
+        let boxed = Box::from_raw(self.struct_value);
+        let arg = Arg::new(&*boxed.as_ptr());
+        // Leak the box since libffi will take ownership of the
```

Comment:
I remember we had discussion on this earlier where it was that said we should detach buffer once used as a struct because it will be freed by libffi AFAICT.